src/app/cli/src/init/client.ml

```diff
@@ -1185,17 +1185,30 @@ let pooled_snapp_commands =
            Yojson.Safe.to_basic
            @@ [%to_yojson: Public_key.Compressed.t option] maybe_public_key
          in
-         let graphql =
-           Graphql_queries.Pooled_snapp_commands.make ~public_key ()
+         let graphql_partial =
+           Graphql_queries.Pooled_snapp_commands_partial.make ~public_key ()
          in
-         let%map response = Graphql_client.query_exn graphql graphql_endpoint in
-         let json_response : Yojson.Safe.t =
-           `List
-             ( List.map
-                 ~f:
-                   (Fn.compose Graphql_client.Snapp_command.to_yojson
-                      Graphql_client.Snapp_command.of_obj)
-             @@ Array.to_list response#pooledSnappCommands )
+         let graphql =
+           object
+             method parse = graphql_partial#parse
```

Comment:
Won't this `parse` method only discover the fields that were present in the original query? It probably makes more sense to generate an OCaml file with the relevant query in it and then let the preprocessor parse the whole thing.

Reply:
oo good idea, this is better